js/h5p-shape.js

```diff
@@ -53,7 +53,148 @@ H5P.Shape = (function ($) {
   };
 
   /**
-   * Style the shape
+   * Is this a svg?
+   * @return {boolean}
+   */
+  C.prototype.isSVG = function () {
+    return this.params.type === 'long-arrow-right' || 
+           this.params.type === 'long-arrow-left' ||
+           this.params.type === 'long-arrow-up' ||
+           this.params.type === 'long-arrow-down' ||
+           this.params.type === 'arrows-alt-h' ||
+           this.params.type === 'arrows-alt-v' ||
+           this.params.type === 'arrows-alt' ||
+           this.params.type === 'triangle' ||
+           this.params.type === 'pentagon' ||
+           this.params.type === 'hexagon' ||
+           this.params.type === 'cylinder' ||
+           this.params.type === 'cube' ||
+           this.params.type === 'cone'
+  };
+
+  /**
+   * Find type and style
+   */
+  C.prototype.findTypeAndStyle = function () {
+    if(this.isSVG()) {
+      this.styleSVG();
+    } else {
+      this.styleShape();
+    }
+  }
+
+  /**
+   * Style a svg
+   */
+  C.prototype.styleSVG = function () {
+    let css = {};
+    let thisType = this.params.type;
+    let uri;
+
+    if(thisType == 'triangle' || thisType == 'pentagon' || thisType == 'hexagon') {
+      uri = this.styleSVGPolygon(thisType);
+    } else if(thisType == 'cylinder' || thisType == 'cube' || thisType == 'cone') {
+      uri = this.styleSVG3d(thisType); 
+    } else {
+      // all other svg's
+      uri = this.getSVGURI(thisType);
+    }
+      const encoded = encodeURIComponent(uri);
+      const backgroundImage = `url("data:image/svg+xml;utf8,${encoded}")`;
+      css['background-image'] = backgroundImage;
+      css['background-size'] = 'cover';
+
+    this.$shape.css(css);
+  }
+
+  /**
+   * Create SVG3d
+   */
+  C.prototype.styleSVG3d = function(type) {
+    let returnURI = '';
+    let borderColor;
+    
+    // in case we only have one 'field' in the semantics, because then the colorSVG becomes a child of this.params.svg3d
+    if (typeof this.params.svg3d === 'string') {
+      borderColor = this.params.svg3d;
+    } else {
+      borderColor = this.params.svg3d.borderColor;
+    }
+    if(type == 'cylinder'){
+      returnURI = `<svg xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="none" viewBox="80 0 355 512"><g fill="${borderColor}"><path d="M425.621,38.187C414.763,1.216,272.789,0,256,0S97.237,1.216,86.379,38.187c-0.64,1.387-1.045,2.859-1.045,4.48v426.667c0,1.621,0.405,3.093,1.045,4.48C97.237,510.784,239.211,512,256,512s158.763-1.216,169.621-38.187c0.64-1.387,1.045-2.859,1.045-4.48V42.667C426.667,41.045,426.261,39.573,425.621,38.187z M256,21.333c87.723,0,137.685,13.248,148.075,21.333C393.685,50.752,343.723,64,256,64S118.315,50.752,107.925,42.667C118.315,34.581,168.277,21.333,256,21.333z M405.333,467.989c-6.101,7.851-56.448,22.677-149.333,22.677c-93.995,0-144.619-15.211-149.333-21.333V65.429C149.312,84.544,242.603,85.333,256,85.333s106.688-0.789,149.333-19.904V467.989z"/></g></svg>`
+    } else if(type == 'cube') {
```

Comment:
Burde svg'ene være lagt ut i egne filer?

Reply:
Noen av svg-stringene har variabler i seg, som farge og tykkelse osv, som blir puttet rett inn i stringen.. Om jeg trekker svg-ene ut, vil jeg da kunne legge inn variablene på en måte?

Reply:
Ikke hvis de er satt inn som bakgrunnsbilde, nei. Har du prøvd å sette svg-ene inn som kode istedenfor? Ikke sikkert det funker, altså. Kan hende vi kunne dratt svg-ene ut som funksjoner i det minste, så vi slipper å ha dem rett i switchen. For eksempel noe sånt (hvis dette ikke blir enda rotete-ere hehehe):

```js
const svgGenerators = (type, borderColor) => {
  return {
    cylinder: (borderColor) => ...,
	cube: (borderColor) => ...,
  }[type]()
};

C.prototype.styleSVG3d = function(type) {
  let borderColor = ...;

  return svgGenerators(type, borderColor);
```